frontend/src/views/WheresMyGene/components/GeneSearchBar/components/QuickSelect/index.tsx

```diff
@@ -252,10 +253,9 @@ export default function QuickSelect<
     }
   };
   const selectedAsArray = Array.isArray(selected) ? selected : [selected];
-
-  const itemOptionsWithSelectedOnTop = [
-    ...new Set([...selectedAsArray, ...items]),
-  ];
+  const itemOptionsWithSelectedOnTop = useMemo(() => {
+    return [...new Set([...selectedAsArray, ...items])];
+  }, [open]);
```

Comment:
This dependency list seems incomplete?

Reply:
Do you have the eslint extension installed to catch React hook bugs like this?

Reply:
It is purposefully incomplete to avoid `itemOptionsWithSelectedOnTop` from being updated when the selectedAsArray changes (though I think `items` is safe to add to the dependency list so I'll add that)

Reply:
Nevermind, adding `items` to dependency list breaks the desired behavior because the list is getting recreated in memory somewhere upstream.